tests/test_feedexport.py

```diff
@@ -1321,6 +1324,420 @@ def test_extend_kwargs(self):
+            result = bz2.decompress(data[filename])
+            self.assertEqual(compressed, data[filename])
+            self.assertEqual(self.expected, result)
+
+    @defer.inlineCallbacks
+    def test_custom_plugin(self):
+
+        filename_to_processed = {
+            self._named_tempfile('newline'): b'foo\r\n\nbar\r\n\n',
+            self._named_tempfile('tab'): b'foo\r\n\tbar\r\n\t',
+        }
+
+        settings = {
+            'FEEDS': {
+                self._named_tempfile('newline'): {
+                    'format': 'csv',
+                    'postprocessing': [self.MyPlugin1],
+                    'plugin1_char': b'\n'
+                },
+                self._named_tempfile('tab'): {
+                    'format': 'csv',
+                    'postprocessing': [self.MyPlugin1],
+                    'plugin1_char': b'\t',
+                },
+            },
+        }
+
+        data = yield self.exported_data(self.items, settings)
+
+        for filename, expected in filename_to_processed.items():
+            self.assertEqual(expected, data[filename])
+
+    @defer.inlineCallbacks
+    def test_custom_plugin_with_compression(self):
+
+        expected = b'foo\r\n\nbar\r\n\n'
+
+        filename_to_decompressor = {
+            self._named_tempfile('bz2'): bz2.decompress,
+            self._named_tempfile('lzma'): lzma.decompress,
+            self._named_tempfile('gzip'): gzip.decompress,
+        }
+
+        settings = {
+            'FEEDS': {
+                self._named_tempfile('bz2'): {
+                    'format': 'csv',
+                    'postprocessing': [self.MyPlugin1, 'scrapy.extensions.postprocessing.Bz2Plugin'],
+                    'plugin1_char': b'\n',
+                },
+                self._named_tempfile('lzma'): {
+                    'format': 'csv',
+                    'postprocessing': [self.MyPlugin1, 'scrapy.extensions.postprocessing.LZMAPlugin'],
+                    'plugin1_char': b'\n',
+                },
+                self._named_tempfile('gzip'): {
+                    'format': 'csv',
+                    'postprocessing': [self.MyPlugin1, 'scrapy.extensions.postprocessing.GzipPlugin'],
+                    'plugin1_char': b'\n',
+                },
+            },
+        }
+
+        data = yield self.exported_data(self.items, settings)
+
+        for filename, decompressor in filename_to_decompressor.items():
+            result = decompressor(data[filename])
+            self.assertEqual(expected, result)
+
+    @defer.inlineCallbacks
+    def test_exports_compatibility_with_postproc(self):
+
+        format_to_expected = {
+            'csv': b'foo\r\nbar\r\n',
+            'json': b'[\n{"foo": "bar"}\n]',
+            'jsonlines': b'{"foo": "bar"}\n',
+            'xml': b'<?xml version="1.0" encoding="utf-8"?>\n<items>\n<item><foo>bar</foo></item>\n</items>',
+            'marshal': b'{\xda\x03foo\xda\x03bar0',
```

Comment:
Even `marshal.dumps` is not giving consistent data. I'll look into this.

Reply:
If the problem is across Python versions, you could use `marshal.dumps({"foo": "bar"})` here instead of hardcoding the value.